### DISCUSSION TITLE
Clojure/Calva mode

### DISCUSSION THREAD

**Discussion by zetashift** [71/dance#212] [Q&A] [answered]
As this is more of a question and not an issue I'm starting a discussion rather than opening a ticket, I hope that's okay!

So I'm looking to set up handy keybindings based a on a `localleader` key like `SPC-m` or `,` so I can do like `SPC-m .` to run the command `calva.jackInOrConnect`, am I right in that would look something like this in the keybindings.json:

``` 
{
    "key": ", ,",
    "command": "calva.jackIn",
    "when": "editorTextFocus && editorLangId == clojure"
}
```

Are there better ways? Is it possible to make a menu out of this?

EDIT: I also can't seem to chord more than 2 keys

**Comment by tshort** [comment, marked as answer]
This sounds like a good role for a menu after you press `,`. Then, `,` - `.` (or whatever) could execute `calva.jackIn`. There are lots of ways to make leader menus. The [wiki](https://github.com/71/dance/wiki) has a number of menu-related entries.

I use a huge leader menu tree via VSpaceCode (see [here](https://github.com/71/dance/wiki/SpaceDance:-leader-menus#vspacecode-menu-tree)). With that, the `,` key (or `space`-`m`) launches a menu based on the current language. I recently sent in a PR to add the Julia language ([here](https://github.com/VSpaceCode/VSpaceCode/pull/251/files)). It works great. 

It's also easy to do this with Dance's builtin menus.

**Reply by zetashift** [reply]
Oh thanks a lot! I haven't looked into menu's a lot yet, how would I do it with Dance's builtin menu? Is it like adding an extra clause to the `when` condition?

**Reply by tshort** [reply]
Use `dance.openWindow("some-window-name")`. Here's an example: https://github.com/71/dance/wiki/Window-menu.

The menu(s) go in `settings.json` within `"dance.menus": {   ...    }`.

**Reply by zetashift** [reply]
Sorry for the late reply, thank you for all the help :D
